packages/wrangler/e2e/c3-integration.test.ts

```diff
@@ -0,0 +1,90 @@
+import crypto from "node:crypto";
+import { existsSync } from "node:fs";
+import path from "node:path";
+import shellac from "shellac";
+import { fetch } from "undici";
+import { beforeAll, describe, expect, it } from "vitest";
+import { normalizeOutput } from "./helpers/normalize";
+import { retry } from "./helpers/retry";
+import { makeRoot } from "./helpers/setup";
+import { WRANGLER } from "./helpers/wrangler-command";
+
+function matchWorkersDev(stdout: string): string {
+	return stdout.match(
+		/https:\/\/smoke-test-worker-.+?\.(.+?\.workers\.dev)/
+	)?.[1] as string;
+}
+
+describe("c3 integration", () => {
+	let workerName: string;
+	let workerPath: string;
+	let workersDev: string | null = null;
+	let runInRoot: typeof shellac;
+	let runInWorker: typeof shellac;
+	let normalize: (str: string) => string;
+
+	beforeAll(async () => {
+		const root = await makeRoot();
+		runInRoot = shellac.in(root).env(process.env);
+		workerName = `smoke-test-worker-${crypto.randomBytes(4).toString("hex")}`;
+		workerPath = path.join(root, workerName);
+		runInWorker = shellac.in(workerPath).env(process.env);
+		normalize = (str) =>
+			normalizeOutput(str, { [workerName]: "smoke-test-worker" });
+	});
+
+	it("init project via c3", async () => {
+		const pathToC3 = path.resolve(__dirname, "../../create-cloudflare");
```

Comment:
Ideally this would use the same `npm pack` approach as Wrangler, but that can be a followup